src/DSE.Open.Values/AgeInMonths.cs

```diff
@@ -1,201 +1,184 @@
 // Copyright (c) Down Syndrome Education International and Contributors. All Rights Reserved.
 // Down Syndrome Education International and Contributors licence this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
 using System.Text.Json.Serialization;
 using DSE.Open.Values.Text.Json.Serialization;
 
 namespace DSE.Open.Values;
 
-[JsonConverter(typeof(JsonStringAgeInMonthsConverter))]
+[JsonConverter(typeof(JsonSpanSerializableValueConverter<AgeInMonths, int>))]
+[ComparableValue]
 [StructLayout(LayoutKind.Auto)]
-public readonly record struct AgeInMonths : ISpanFormattable, ISpanParsable<AgeInMonths>, IComparable<AgeInMonths>
+public readonly partial struct AgeInMonths : IComparableValue<AgeInMonths, int>
 {
-    private const string DefaultFormat = "0";
+    private const int MaxYears = int.MaxValue / 12;
+    private const int MaxMonths = int.MaxValue;
 
     public static readonly AgeInMonths Zero;
 
-    public AgeInMonths(int years, int months) : this((years * 12) + months)
+    public static int MaxSerializedCharLength => 12;
+
+    public AgeInMonths(int years, int months)
     {
+        ArgumentOutOfRangeException.ThrowIfNegative(years);
+        ArgumentOutOfRangeException.ThrowIfNegative(months);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(years, MaxYears);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(months, MaxMonths);
+
+        var maxMonths = int.MaxValue - (years * 12);
+
+        if (months > maxMonths)
+        {
+            throw new ArgumentOutOfRangeException(nameof(years),
+                "The resulting value is outside the range of a AgeInMonths");
+        }
+
+        checked
+        {
+            TotalMonths = (years * 12) + months;
+        }
     }
 
     public AgeInMonths(int totalMonths)
     {
+        ArgumentOutOfRangeException.ThrowIfNegative(totalMonths);
         TotalMonths = totalMonths;
     }
 
     /// <summary>
-    /// The number of complete years represented by the value.
+    /// The total number of months represented by the value.
     /// </summary>
-    public int Years => TotalMonths / 12;
+    public int TotalMonths { get; }
 
     /// <summary>
-    /// The number of months represented by the value in addition to the <see cref="Years"/>.
+    /// The total number of years and months represented by the value.
     /// </summary>
-    public int Months => TotalMonths % 12;
```

Comment:
From an API use perspective, it felt like `Months` should be returning `TotalMonths`, so to make it clearer that it is the remainder of dividing by `Years` I went with a single method returning a tuple.

Reply:
I think I was aiming to follow the pattern here: https://learn.microsoft.com/en-us/dotnet/api/system.timespan?view=net-8.0#properties and https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs,170